internal/pkg/utils/deepcopy/deepcopy.go

```diff
@@ -125,13 +125,19 @@ func translateRecursive(clone, original reflect.Value, callback TranslateFunc, s
 	case kind == reflect.Map:
 		if !original.IsNil() {
 			clone.Set(reflect.MakeMap(originalType))
-			for _, key := range original.MapKeys() {
-				originalValue := original.MapIndex(key)
+			for _, originalKey := range original.MapKeys() {
+				// Clone key
+				cloneKey := reflect.New(originalKey.Type()).Elem()
+				keySteps := steps.Add(MapKeyValueStep{key: originalKey.Interface()})
+				translateRecursive(cloneKey, originalKey, callback, keySteps, visitedPtr)
```

Comment:
`deepcopy.CopyTranslate` metoda doteraz neumoznovala nahradit kluc, potrebujem to, tak som to doplnil.
Tu sa riesi `translate` kluca v go native map.